saltedge-compliance-connector-java-v2/src/main/java/com/saltedge/connector/example/controllers/consent/UserConsentController.java

```diff
@@ -160,6 +165,7 @@ public ModelAndView onPaymentConfirmationSubmit(
         extra = payment.extra;
         payment.status = PaymentStatus.FAILED;
         paymentsRepository.save(payment);
+        if (account != null) updatePaymentFundsInformation(payment, account.availableAmount);
```

Comment:
Invalid solution. It has no sense to update funds information if payment has failed status. Please think before doing.